src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,189 @@
+        } else if (userInput.length() > 4 && (userInput.substring(0, 4)).equals("todo")) {
+            addToDo(userInput);
+            storage.save(tasklist);
+        } else if (userInput.equals("todo")) {
+            throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+        } else if (userInput.equals("deadline")) {
+            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+        } else if (userInput.equals("event")) {
+            throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+        } else if (userInput.length() > 8 && (userInput.substring(0, 8)).equals("deadline")) {
+            addDeadline(userInput);
+            storage.save(tasklist);
+        } else if (userInput.length() > 5 && (userInput.substring(0, 5)).equals("event")) {
+            addEvent(userInput);
+            storage.save(tasklist);
+        } else if (userInput.length() > 7 && (userInput.substring(0, 6)).equals("delete")) {
+            deleteTask(userInput);
+            storage.save(tasklist);
+        } else if (userInput.startsWith("find ")) {
+            findTask(userInput);
+        } else {
+            System.out.println("I'm sorry, but I don't know what that means! Try typing something else!");
+        }
+    }
+
+    /**
+     * Displays list of tasks to user.
+     */
+    public static void showList(){
+        ui.printList(tasklist.listOfTasks());
+    }
+
+    /**
+     * Marks tasks as done.
+     *
+     * @param task The string containing which task to be marked.
+     */
+    public static void markDone(String task) {
+        int taskToMark = 0;
+        String strTaskToMark = "";
+
+        for (int j = 5; j < task.length(); j++) {
+            strTaskToMark = strTaskToMark + task.charAt(j);
+        }
+        taskToMark = Integer.parseInt(strTaskToMark);
+        ui.printDone(tasklist.mark(taskToMark - 1));
+    }
+
+    /**
+     * Changes status of task back to not done.
+     *
+     * @param task The string containing which task to be unmarked.
+     */
+    public static void markUndone(String task) {
+        int taskToUnmark = 0;
+        String strTaskToUnmark = "";
+
+        for (int j = 7; j < task.length(); j++) {
+            strTaskToUnmark = strTaskToUnmark + task.charAt(j);
+        }
+
+        taskToUnmark = Integer.parseInt(strTaskToUnmark);
+        ui.printUndone(tasklist.unmark(taskToUnmark - 1));
+    }
+
+    /**
+     * Adds Todo tasks.
+     *
+     * @param str The string containing task to be added.
+     */
+    public static void addToDo(String str) {
+        ToDos newToDo = new ToDos(str.substring(5, str.length()));
+        ui.printTodo(tasklist.addTask(newToDo));
+        ui.printTasksLeft(tasklist.getSize());
+    }
+
+    /**
+     * Adds Deadline tasks.
+     *
+     * @param str The string containing task to be added.
+     */
+    public static void addDeadline(String str) {
+        int k = 9;
```

Comment:
Perhaps move this to above line 121 so that it is clear this is a loop counter?

It's also unclear to me why the counter starts at 9.

A similar thing applies to `addEvent`.